Update quickstart doc

Chaho12 opened the pull request:
- Udate quickstart doc to version ~5~ 6 with db checking code
- add var to gitignore in all directory hierarchy (for local dev) like trino https://github.com/trinodb/trino/commit/45bf374d6139406a83472ccd9314381a3280d955

mosabua commented on docs/quickstart.md:
```diff
@@ -38,6 +38,8 @@ two Trino servers and a PostgreSQL database running in Docker, and a Trino
 have the `brew` package manager installed, this script attempts to install
 `psql`, if not you must install `psql` through other means.
+- If you use `brew` package, you might have to set libpq first in your PATH, run:
+- `echo 'export PATH="/opt/homebrew/opt/libpq/bin:$PATH"' >> ~/.zshrc`
```
This assume usage of zsh ... dont do that. Just say what folder has to be added to the PATH. Something like 

Use the following command in your shell startup script or manually: 

```
export PATH="/opt/homebrew/opt/libpq/bin:$PATH"
```

mosabua commented on docs/quickstart.md:
```diff
@@ -65,15 +67,20 @@ if ! $(command -v psql >/dev/null); then
-psql  -U postgres -h localhost -d gateway -f ./gateway-ha-persistence-postgres.sql 
+#Check if DB is running
+if docker ps --format '{{.Names}}' | grep -q '^local-postgres$'; then
+    echo "Postgres DB 'localhost-postgre' container is already running, so only run gateway server."
```
```suggestion
    echo "PostgreSQL database container 'localhost-postgre' is already running. Only starting Trino Gateway."
```

Chaho12 replied:
Fixed :)

ebyhr commented on docs/quickstart.md:
```diff
@@ -37,7 +37,10 @@ The scripts from this quickstart guide set up a local environment consisting of
-`psql`, if not you must install `psql` through other means.
+`psql`, if not you must install `psql` through other means. 
+
+Use the following command in your shell startup script or manually:
+- `export PATH="/opt/homebrew/opt/libpq/bin:$PATH"`
```
This section should be rewritten to use psql in a container instead of local psql. Suggesting brew install doesn't make sense as it already exists in the container. 

Also, we already have TrinoGatewayRunner that initialize the database automatically. I feel this section is increasing managing costs with less benefits.

Chaho12 replied:
I think this section is for users to play with gateway without any mvn builds, just use public gateway image.


> This section should be rewritten to use psql in a container instead of local psql. Suggesting brew install doesn't make sense as it already exists in the container.

Now that you point out, we can change this by running psql within container. What do you think?
```bash
# before
psql -U postgres -h localhost -c 'CREATE DATABASE gateway'
psql  -U postgres -h localhost -d gateway -f gateway-ha/src/main/resources/gateway-ha-persistence-postgres.sql
# after
docker exec local-postgres psql -U postgres -h localhost -c 'CREATE DATABASE gateway'
docker cp gateway-ha/src/main/resources/gateway-ha-persistence-postgres.sql local-postgres:/tmp
docker exec local-postgres psql -U postgres -h localhost -d gateway -f /tmp/gateway-ha-persistence-postgres.sql

```

Chaho12 replied:
@willmostly any thoughts?

ebyhr replied:
> I think this section is for users to play with gateway without any mvn builds, just use public gateway image with postgre image

Mentioning docker compose yml of `docker` directory should be sufficient. We should avoid managing duplicated information.

willmostly replied:
This predated the gateway docker image, so its worthwhile to consider if a play environment that uses the bare jars is still worthwhile to maintain. However as @mosabua says below we should evaluate this as part of a strategic reorganization.

ebyhr replied:
This command downloads jar from maven central. It means no-difference from using docker container. 
I still disagree with having this commands in quickstart.

mosabua replied:
We have to pull this apart for quick start with docker and quick start with jar as soon as possible. However that does not necessarily have to be part of this PR.

mosabua commented:
Just to clarify .. I think we will need to clearly separate the different usage aspects: 

* developer vs platform admin tasks
* using jar vs docker vs helm

Currently its all mixed about and thats the root cause of the problem. Like I mentioned in the last dev sync I am hoping to get a doc site going and then break up the docs around all these aspects. With that in mind .. lets keep this PR simple so we can merge and do follow ups as needed.

Chaho12 commented:
As I moved psql command to be executed within postgresql docker, user does now does not need to have homebrew or libpq.
So I cleaned up bash script for easier use.

willmostly commented on docs/quickstart.md:
```diff
@@ -50,30 +48,31 @@ PostgreSQL database at localhost:5432.
+
+    #Initialize the DB
+    docker exec local-postgres psql -U postgres -h localhost -c 'CREATE DATABASE gateway'
+    docker cp ../gateway-ha/src/main/resources/gateway-ha-persistence-postgres.sql local-postgres:/tmp
```
could you start the container with `docker run -v ../gateway-ha/src/main/resources/gateway-ha-persistence-postgres.sql:/tmp/gateway-ha-persistence-postgres.sql ...` instead of copying?

Chaho12 replied:
changed to use mount volume command :)

willmostly reviewed (approved):
Looks good, modulo @mosabua 's comments on overall organization.

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
-psql  -U postgres -h localhost -d gateway -f ./gateway-ha-persistence-postgres.sql 
+#Check if DB is running
+if docker ps --format '{{.Names}}' | grep -q '^local-postgres$'; then
+    echo "PostgreSQL database container 'localhost-postgre' is already running. Only starting Trino Gateway."
```
localhost-postgres ?

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
-    fi
+#Check and get the Gateway Jar
+if [[ -f "gateway-ha.jar" ]]; then
+    echo "Trino Gateway Jar already exists so skip fetching."
```
Maybe better to say

Found gateway-har.jar file in current directory.

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
+if [[ -f "gateway-ha.jar" ]]; then
+    echo "Trino Gateway Jar already exists so skip fetching."
+else
+    echo "Trino Gateway Jar does not exist, try to fetch from maven repository."
```
Failed to find gateway-ha.jar in current directory. Fetching version $VERSION from Maven Central repository.

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
+if docker ps --format '{{.Names}}' | grep -q '^local-postgres$'; then
+    echo "PostgreSQL database container 'localhost-postgre' is already running. Only starting Trino Gateway."
+else
+    echo "PostgreSQL database container 'localhost-postgre' is not running, initialize and run database"
```
```suggestion
    echo "PostgreSQL database container 'localhost-postgre' is not running. Proceeding to initialize and run database server."
```

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
+fi
 
-#Start the DB
+#Start the Gateway server. Please check #Build and run in development doc if you don't have jar file built yet.
```
```suggestion
#Start Trino Gateway server.
```

Chaho12 commented:
> Currently its all mixed about and thats the root cause of the problem. Like I mentioned in the last dev sync I am hoping to get a doc site going and then break up the docs around all these aspects. With that in mind .. lets keep this PR simple so we can merge and do follow ups as needed.

rebased and fixed wordings

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
-        brew install libpq
-    fi
+#Check and get the Gateway Jar
+if [[ -f "gateway-ha.jar" ]]; then
```
this checks in the current folder of the script

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
+else
+    echo "PostgreSQL database container 'localhost-postgres' is not running. Proceeding to initialize and run database server."
+    export PGPASSWORD=mysecretpassword
+    docker run -v ./gateway-ha/src/main/resources/gateway-ha-persistence-postgres.sql:/tmp/gateway-ha-persistence-postgres.sql --name local-postgres -p 5432:5432 -e POSTGRES_PASSWORD=$PGPASSWORD -d postgres:latest
```
this assumes the project checkout root folder `trino-gateway` is the location of the quickstart script

mosabua commented on docs/quickstart.md:
```diff
@@ -50,32 +48,32 @@ PostgreSQL database at localhost:5432.
 
-#Start the DB
+#Start Trino Gateway server.
 java -Xmx1g --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED -jar ./gateway-ha.jar server ./quickstart-config.yaml
```
this assumes the script is run in the docs folder .. or the config is copied

mosabua reviewed (changes requested):
I tried this now and as it stands the script does not work. Or at least it is not clear to me where the script should be created and run from. See my comments .. 

We should add the info on what files are needed where and then make it work for that setup. 

In the longer run we have to separate dev, testing, and productions setup docs .. but for now this is fine .. as long as it actually works and can be followed. Please work together @willmostly and @Chaho12 to figure out what direction you want to take in terms of specified folder. Maybe it would be even cleaner if we have a list of files to copy into a random new folder.

mosabua reviewed (approved):
Looks good now. Thanks for addressing the comments.
